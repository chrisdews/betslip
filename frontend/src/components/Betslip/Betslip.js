import React, { useEffect, useState } from "react";
import Header from "../Header";
import BetsContainer from "../BetsContainer";
import ReceiptModal from "../ReceiptModal";

function Betslip() {
  const [bets, setBets] = useState(null);
  const [stakes, setStakes] = useState({});
  const [betPlaced, setBetPlaced] = useState(false);
  const [betButtonDisabled, setBetButtonDisabled] = useState(true);
  const [oddsFilter, setOddsFilter] = useState("all");

  useEffect(() => {
    getBetInfo();
  }, [oddsFilter]);

  const getBetInfo = async () => {
    await fetch(`http://localhost:4000/${oddsFilter}`, {
      method: "GET",
    })
      .then((res) => res.json())
      .then((res) => {
        setBets(res.filteredData);
      })
      .catch((err) => {
        console.error(err);
      });
  };

  const betBuilder = (betName, stake) => {
    setStakes((stakes) => ({
      ...stakes,
      [betName]: stake,
    }));
    setBetButtonDisabled(false);
  };

  return betPlaced ? (
    <ReceiptModal stakes={stakes} />
  ) : (
    <span data-testid="bets-list">
      <Header setOddsFilter={setOddsFilter} />
      <BetsContainer
        bets={bets}
        betBuilder={betBuilder}
        setBetPlaced={setBetPlaced}
        betButtonDisabled={betButtonDisabled}
      />
    </span>
  );
}

export default Betslip;
