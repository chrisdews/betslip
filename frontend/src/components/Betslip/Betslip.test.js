import React from "react";
import Betslip from "./Betslip";
import { render, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom";

describe("Betslip", () => {
  it("renders correctly - snapshot", () => {
    const { container } = render(<Betslip />);
    expect(container).toMatchSnapshot();
  });

  it("the header component should be rendered on initial load", () => {
    const { getByText } = render(<Betslip />);
    expect(getByText(/betslip/i)).not.toBe(null)
  });

  it("the Loading component should be rendered on initial load", () => {
    const { getByText } = render(<Betslip />);
    expect(getByText(/loading.../i)).not.toBe(null)
  });

  it("should display the bets component after successful api call", async () => {
    const { getByTestId } = render(<Betslip />);
    const betsList = await waitFor(() => getByTestId(/bets-list/i))
    expect(betsList).toBeInTheDocument()
  });
});

// add tests for the filter (loop through the bets list and check values)