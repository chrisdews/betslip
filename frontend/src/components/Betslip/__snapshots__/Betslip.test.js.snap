// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Betslip renders correctly - snapshot 1`] = `
<div>
  <span
    data-testid="bets-list"
  >
    <div
      class="sc-dlnjwi oaTBY"
    >
      <div
        class="sc-jSFjdj dqnCUv"
      >
        <h1
          class="sc-eCApnc bGVqgi"
        >
          BETSLIP
        </h1>
        <img
          class="sc-hKFxyN eaodwR"
          src="oc-logo.png"
        />
      </div>
      <select
        class="sc-bdnxRM einyrw"
        id="selectList"
        name="selectList"
      >
        <option
          class="sc-gtsrHT DFeHs"
          data-testid="option-all"
          value="all"
        >
          ALL
        </option>
        <option
          class="sc-gtsrHT DFeHs"
          data-testid="option-less-than-2"
          value="decimalOddsLessThanTwo"
        >
          Odds less than 2
        </option>
        <option
          class="sc-gtsrHT DFeHs"
          data-testid="option-more-than-2"
          value="decimalOddsMoreThanTwo"
        >
          Odds greater than 2
        </option>
      </select>
    </div>
    <div
      class="sc-jrsJWt cgsxPM"
    >
      <h3>
        loading...
      </h3>
      <img
        class="sc-pNWdM letrpD"
        src="oc-logo.png"
      />
    </div>
  </span>
</div>
`;
